.github/workflows/ci.yml

```diff
@@ -21,6 +21,8 @@ jobs:
     steps:
       - name: Checkout Repo
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 ## Needed for "Set internal/tools/go.mod timestamp" step.
```

Comment:
I think I'm missing something here. How is fetching all history for all branches and tags needed to set timestamps?

Reply:
```
          filename="internal/tools/go.mod"
          unixtime=$(git log -1 --format="%at" -- "${filename}")
          touchtime=$(date -d @$unixtime +'%Y%m%d%H%M.%S')
          touch -t ${touchtime} "${filename}"
          ls -la --time-style=full-iso "${filename}"
```

This `git log -1 --format="%at" -- "${filename}` was not returning proper timestamp.